### PR TITLE
[RFC][NI] Fix uni-modal observer first run

### PR DESCRIPTION
**Pull request by migueldemoura**
Fixes the first run flag logic and adds some tests for this behaviour.

### CODE REVIEW

**Review by mmelo — APPROVED** [review]

File: addon/components/uni-modal.js [hunk 1]
```diff
@@ -19,27 +19,32 @@ export default Component.extend({
...
+  setBodyOverflowClass(state) {
+    let body = document.querySelector('body');
+    state
+      ? body.classList.add(this.get('bodyOverflowClass'))
+      : body.classList.remove(this.get('bodyOverflowClass'));
+  },
+
   // This observer is used to bypass the scroll on mobile when a modal is open
   onOpenChangeObserver: on('init', observer('isOpen', function() {
-    if (this.get('firstLoad') && !this.get('isOpen')) {
```
Comment (AndreJoaquim) [inline addon/components/uni-modal.js:24]: Booleans start with a verb: `firstLoad` => `isFirstLoad`

File: addon/components/uni-modal.js [hunk 2]
```diff
@@ -19,27 +19,32 @@ export default Component.extend({
...
+    if (this.get('firstLoad')) {
```
Comment (AndreJoaquim) [inline addon/components/uni-modal.js]: @AndreJoaquim  (@migueldemoura  **edit**: cough cough)
```js
let firstLoad = this.get('firstLoad');
this.set('firstLoad', false);

if (firstLoad && !this.get('isOpen')) {
  return;
}

this.setBodyOverflowClass(this.get('isOpen'));
```
@satcin 
```js
if (!this.get('firstLoad')) {
  this.setBodyOverflowClass(this.get('isOpen'));

  return;
}

this.set('firstLoad', !this.get('firstLoad'));
if (!this.get('isOpen')) {
  return;
}

this.setBodyOverflowClass(this.get('isOpen'));
```

Comment (AndreJoaquim) [reply addon/components/uni-modal.js]: It was fun 😄

**Review by AndreJoaquim — CHANGES_REQUESTED** [review]

File: addon/components/uni-modal.js (hunk 2 above)
Comment (migueldemoura) [reply addon/components/uni-modal.js]: Top pair programming 😆 😆 😆 . @satcin, hummm

**Review by AndreJoaquim — APPROVED** [review]

**Review by AndreJoaquim — APPROVED** [review]

**Pull request merged**